tests/mock_s3_server/mock_s3_server.py

```diff
@@ -264,7 +305,7 @@ def handle_get_object_modified(start_range, end_range, request):
     data_length = end_range - start_range
 
     if start_range == 0:
-        return "/get_object_modified_first_part", data_length, True
+        return ResponseConfig("/get_object_modified_first_part", data_length)
```

Comment:
Since you're just passing positional args, this is going to end up setting `disconnect_after_headers=data_length`
```suggestion
        return ResponseConfig("/get_object_modified_first_part", generate_body_size=data_length)
```